postgres/src/test/scala/zio/sql/postgresql/FunctionDefSpec.scala

```diff
@@ -151,7 +151,105 @@ object FunctionDefSpec extends PostgresRunnableSpec with ShopSchema {
         } yield assert(r.head)(equalTo(expected))
 
         assertion.mapErrorCause(cause => Cause.stackless(cause.untraced))
-      }
+      },
+      suite("format function")(
+        testM("format0") {
+          import Expr._
+
+          val query = select(Format0("Person")) from customers
+
+          val expected = Seq(
+            "Person",
+            "Person",
+            "Person",
+            "Person",
+            "Person"
+          )
+
+          val testResult = execute(query.to[String, String](identity))
+          collectAndCompare(expected, testResult)
+        },
+        testM("format1") {
+          import Expr._
+
+          val query = select(Format1("Person: %s", Customers.fName)) from customers
+
+          val expected = Seq(
+            "Person: Ronald",
+            "Person: Terrence",
+            "Person: Mila",
+            "Person: Alana",
+            "Person: Jose"
+          )
+
+          val testResult = execute(query.to[String, String](identity))
+          collectAndCompare(expected, testResult)
+        },
+        testM("format2") {
+          import Expr._
+
+          val query = select(Format2("Person: %s %s", Customers.fName, Customers.lName)) from customers
+
+          val expected = Seq(
+            "Person: Ronald Russell",
+            "Person: Terrence Noel",
+            "Person: Mila Paterso",
+            "Person: Alana Murray",
+            "Person: Jose Wiggins"
+          )
+
+          val testResult = execute(query.to[String, String](identity))
+          collectAndCompare(expected, testResult)
+        },
+        testM("format3") {
+          import Expr._
+
+          val query = select(Format3("Person: %s %s with double quoted %I ", Customers.fName, Customers.lName, "identi fier")) from customers
+
+          val expected = Seq(
+            s"""Person: Ronald Russell with double quoted "identi fier" """,
+            s"""Person: Terrence Noel with double quoted "identi fier" """,
+            s"""Person: Mila Paterso with double quoted "identi fier" """,
+            s"""Person: Alana Murray with double quoted "identi fier" """,
+            s"""Person: Jose Wiggins with double quoted "identi fier" """
+          )
+
+          val testResult = execute(query.to[String, String](identity))
+          collectAndCompare(expected, testResult)
+        },
+        testM("format4") {
+          import Expr._
+
+          val query = select(Format4("Person: %s %s with null-literal %L and non-null-literal %L ", Customers.fName, Customers.lName, "FIXME: NULL", "literal")) from customers
```

Comment:
couldn't find a solution to specify NULL
tried `Option.empty[String]` and `null.asInstanceOf[String]`

Reply:
Yeah, Options are not well supported at this point.